ha/core/event_analyzer/filter/filter.py

```diff
@@ -86,3 +104,52 @@ def filter_event(self, msg: str) -> bool:
         except Exception as e:
             raise EventAnalyzerError(f"Failed to filter event. Message: {msg}, Error: {e}")
 
+
+class IEMFilter(Filter):
+    """ Filter IEM consumed by watcher """
+
+    def __init__(self):
+        """
+        Init method
+        """
+        super().__init__()
+        # Loads IEM filter rules in the configuration
+        ConfigManager.load_filter_rules()
+
+        # Get filter type and resource types list from the IEM rule file
+        self.filter_type = Conf.get(const.ALERT_FILTER_INDEX, "iem.filter_type")
+        self.module_types_list = Conf.get(const.ALERT_FILTER_INDEX, "iem.component.ha.module")
+
+    def filter_event(self, msg: str) -> bool:
+        """
+        Filter event.
+        Args:
+            msg (str): Msg
+        """
+        try:
+            iem_required = False
+            message = json.loads(msg)
+
+            msg_type = message['message']['actuator_response_type']
+            if msg_type is not None:
+                return iem_required
+
+            if not Filter.is_iem_msg(message):
+                return iem_required
+
+            msg_type = message['message']['sensor_response_type']
+            _module_type = msg_type['specific_info']['module']
+
+            if self.filter_type == const.INCLUSION:
```

Comment:
we also need to check component should be in list

also can we modify rule as 
```
    "iem": {
        "filter_type":"inclusion",
        "components": ["ha"],
        "ha": {
            "module": ["Node", "Service"]
        }
    }
```
first match component then module

Reply:
Done, updated the iem filter rules format
"iem": {
    "filter_type":"inclusion",
    "components": ["ha"],
    "modules": {"ha": ["Node", "Service"] }
}